From 5b73b64af7aad348704c735ac38c947220c42478 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Thu, 10 Aug 2023 16:06:12 +0200
Subject: [PATCH 0464/1141] dt-bindings: i3c: Document the STM32 I3C controller

STM32 I3C supports both controller and target role, SDR-only, and
implements all the feature required by the MIPI I3C v1.1 specification.

It requires at least a register mapping, a clock and an interrupt.

Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: Ie634b15584f5839f97f7d634f5cc12a2385176d4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/322413
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../bindings/i3c/st,stm32-i3c-master.yaml     | 53 +++++++++++++++++++
 1 file changed, 53 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/i3c/st,stm32-i3c-master.yaml

diff --git a/Documentation/devicetree/bindings/i3c/st,stm32-i3c-master.yaml b/Documentation/devicetree/bindings/i3c/st,stm32-i3c-master.yaml
new file mode 100644
index 000000000000..6b6331316a6a
--- /dev/null
+++ b/Documentation/devicetree/bindings/i3c/st,stm32-i3c-master.yaml
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/i3c/st,stm32-i3c.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: STM32 I3C master
+
+maintainers:
+  - Amelie Delaunay <amelie.delaunay@foss.st.com>
+
+allOf:
+  - $ref: "i3c.yaml#"
+
+properties:
+  compatible:
+    const: st,stm32-i3c
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    maxItems: 1
+
+  clocks:
+    maxItems: 1
+
+  resets:
+    maxItems: 1
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - clocks
+
+additionalProperties: true
+
+examples:
+  - |
+    #include <dt-bindings/clock/stm32mp25-clks.h>
+    #include <dt-bindings/interrupt-controller/arm-gic.h>
+    #include <dt-bindings/reset/stm32mp25-resets.h>
+    i3c@40190000 {
+      #address-cells = <3>;
+      #size-cells = <0>;
+      compatible = "st,stm32-i3c";
+      reg = <0x40190000 0x400>;
+      interrupts = <GIC_SPI 109 IRQ_TYPE_LEVEL_HIGH>;
+      clocks = <&rcc CK_KER_I3C1>;
+      resets = <&rcc I3C4_R>;
+      status = "disabled";
+    };
\ No newline at end of file
-- 
2.39.2

